src/main/java/com/vmware/vipclient/i18n/messages/service/ComponentService.java

```diff
@@ -117,8 +166,13 @@ public MessageCacheItem getMessages(Iterator<Locale> fallbackLocalesIter) {
     			// Refresh the cacheItem in a separate thread
     			refreshCacheItemTask(cacheItem);
     		}
-    	} else { // Item is not in cache. Create and store cacheItem for the requested locale
-    		cacheItem = createCacheItem(fallbackLocalesIter);
+    	} else { // Item is not in cache.
+			if (waitForFetch) {
+				cacheItem = this.createCacheItem(fallbackLocalesIter);
+			} else {
+				// Create and store cacheItem for the requested locale in a separate thread
+				this.createCacheItemTask(fallbackLocalesIter);
```

Comment:
Looks like this line will never be executed.
Because when it's is first fetch, will execute line 171; when it's not first fetch, will execute line 160 or 167.
 So add 'waitForFetch' flag is no use.

Reply:
The return of getMessagesFromCache can be that of a fallback locale. This is to avoid performance hit.
This is the case where it will enter the else. 

I will update the code in a more readable way.